nflog/nflog.go

```diff
@@ -415,6 +415,11 @@ func (l *Log) Log(r *pb.Receiver, gkey string, firingAlerts, resolvedAlerts []ui
 		}
 	}
 
+	expiresAt := now.Add(l.retention)
+	if expiry > 0 && l.retention > expiry {
```

Comment:
wouldn't it make sense to use the default retention only if expiry > 0?
If repeat_interval is 10d (because why not), it would make sense to keep the log for longer than the default 120h retention. We could also get rid of this check:

https://github.com/prometheus/alertmanager/blob/d034f116d52912f00e444b1b07bb0e97a168074e/cmd/alertmanager/main.go#L488-L500